### PR TITLE
fix(backup): update to match new sstable naming format

### PR DESCRIPTION
**Pull request by dkropachev**
Update code to match new SSTable format "me-3g7k_098r_4wtqo2asamoc1i8h9n-big-CRC.db"

### CODE REVIEW

File: pkg/service/backup/restore_worker_test.go
Comment (tchaikov) [inline pkg/service/backup/restore_worker_test.go:41]:
```suggestion
	for _, tcase := range testCases {
```

**Review by tchaikov — COMMENTED** [review]
thank you for fixing this!

File: pkg/service/backup/restore_worker_test.go
Comment (dkropachev) [reply pkg/service/backup/restore_worker_test.go:41]: it is run in parallel, since tcase is a loop variable it is going to belong to upper context and be the same for every instance of the test.

**Comment by Michal-Leszczynski** [comment]
Which Scylla version introduces this new SSTable ID generation? 
It would be good to include this information in SM restore [compatibility matrix](https://manager.docs.scylladb.com/stable/restore/compatibility-matrix.html), so that users know that SM 3.1 won't work with newer Scylla and that they should upgrade it.

**Comment by tchaikov** [comment]
> Which Scylla version introduces this new SSTable ID generation? It would be good to include this information in SM restore [compatibility matrix](https://manager.docs.scylladb.com/stable/restore/compatibility-matrix.html), so that users know that SM 3.1 won't work with newer Scylla and that they should upgrade it.

@Michal-Leszczynski the version with this new SSTable identifier hasn't released yet. this change was introduced by https://github.com/scylladb/scylladb/commit/f014ccf36962135889a84cff15b0478d711b2306 . since 5.3 has been branch has been created. my guess this change will be included in 5.4, if it will not be reverted again by then =)

**Comment by dkropachev** [comment]
> > Which Scylla version introduces this new SSTable ID generation? It would be good to include this information in SM restore [compatibility matrix](https://manager.docs.scylladb.com/stable/restore/compatibility-matrix.html), so that users know that SM 3.1 won't work with newer Scylla and that they should upgrade it.
> 
> @Michal-Leszczynski the version with this new SSTable identifier hasn't been released yet. this change was introduced by [scylladb/scylladb@f014ccf](https://github.com/scylladb/scylladb/commit/f014ccf36962135889a84cff15b0478d711b2306) . since 5.3 branch has been created. my guess is that this change will be included in 5.4, if it will not be reverted again by then =)

Thanks Kefu.
@Michal-Leszczynski, let's backport it to 3.1 ? And update the matrix

**Comment by Michal-Leszczynski** [comment]
Shouldn't we wait with merging/backporting that until we are sure that this version of regex ends up in some Scylla release?

The other thing, which might be affected by this change, is `schemaWorker`:
https://github.com/scylladb/scylla-manager/blob/fd5532894a98a8600f33be1b82be29421a4bb857/pkg/service/backup/restore_worker_schema.go#L18-L23

https://github.com/scylladb/scylla-manager/blob/fd5532894a98a8600f33be1b82be29421a4bb857/pkg/service/backup/restore_worker_schema.go#L264-L278

Because it's not possible to alter `tombstone_gc` of `system_schema` tables, SM simulates immediate repair by downloading ALL backed-up SSTables into EACH hosts (data duplication, but it should be fine as schema files are rather small). Because SSTables backed-up from different hosts might have the same SSTable ID (they used to be simple, consecutive numbers), SM changes SSTable IDs when downloading them into upload dir, so that we can avoid name conflicts and stream all downloaded data at one time.

Because SSTable IDs used to be simple, consecutive numbers, SM also renames them is such fashion. If this regex change goes through, perhaps it would be good to change the rename ID procedure described above as well.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
or, we might need to check the `uuid_sstable_identifiers_enabled` setting of the target host, and use the integer if it is false, and generated a uuid if it is true. as an alternative, to "remember" the type of the sstable identifier, and generate a new one if it is an integer, and reuse it if it is already a uuid?

**Comment by dkropachev** [comment]
@Michal-Leszczynski , could you please take a look, I have added logic to respect `uuid_sstable_identifiers_enabled`

**Review by karol-kokoszka — APPROVED** [review]
I'm fine with merging it to master.
I don't think we should backport it to previous versions.

We should update the 3.1 matrix (or backport) when Scylla 5.3 is released.

**Comment by dkropachev** [comment]
@Michal-Leszczynski , you have a final word, WDYT, are we merging it ? are you ok with recent changes?

**Comment by Michal-Leszczynski** [comment]
Sorry for such a late response.
My goal was to fix restore tests before reviewing and merging this change (I think I'm ok with it being merged even without confirmed Scylla release).
Can we wait with that a little bit longer?

File: swagger/scylla_v2.json
Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: Idea: maybe we should add gh action step to run:
- `cd swagger`
- `make clean`
- `make`
- check if anything has changed (using git)

Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: @karol-kokoszka

Comment (karol-kokoszka) [reply swagger/scylla_v2.json:4962]: > Off-topic: regenerating all swagger files changes comments styling (example below) in about 1200, so we should perhaps regenerate all of them.

It doesn't if you use correct version of swagger.
It's here https://github.com/scylladb/scylla-manager/blob/88fd44597af881079d47970d215f64dbf967e3f7/install-dependencies.sh#L45

> Idea: maybe we should add gh action step to run:

> cd swagger
> make clean
> make
> check if anything has changed (using git)

Why not. Good idea.

File: pkg/scyllaclient/config_client.go
Comment (Michal-Leszczynski) [reply pkg/scyllaclient/config_client.go]: But still, why would we check for `alternatorDisabled(err)` here?
I know that it is just a cover for checking status 400, but maybe it would be better to check for that in a more direct way.

File: swagger/scylla_v2.json
Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: But in the commit `feat(scyllaclient): add api for config uuid_sstable_identifiers_enabled` we can still see that `gen/agent/models/node_info` was modified while `agent.json` wasn't. After that it is indeed regenerated in a commit which `agent.json` is modified.

Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: Also, looks like import:
```go
"github.com/scylladb/scylla-manager/swagger/gen/scylla-manager/client/operations"
```
is broken in `scylla_manager_clinet.go` (missing previously present v3 in the path).

File: pkg/sstable/sstable_naming.go
Comment (Michal-Leszczynski) [reply pkg/sstable/sstable_naming.go]: Shouldn't the old `regexLaMx` be kept unchanged as:
```go
regexLaMx = regexp.MustCompile(`(la|m[cde])-(\d+)-(\w+)-(.*)`)
```

File: pkg/sstable/sstable_naming.go
Comment (Michal-Leszczynski) [inline pkg/sstable/sstable_naming.go:124]: Maybe it would be good to include a link to an actual place in Scylla where those IDs are generated? This line looks suspicious without any context:
```go
return string(out[0:4]) + "_" + string(out[4:8]) + "_" + string(out[8:])
```
https://github.com/scylladb/scylladb/blob/decbc841b749d8dd3e2ddd4be4817c57d905eff2/sstables/generation_type.hh#L68

File: pkg/scyllaclient/config_client.go
Comment (dkropachev) [reply pkg/scyllaclient/config_client.go]: Could you please elaborate

File: swagger/scylla_v2.json
Comment (dkropachev) [reply swagger/scylla_v2.json:4962]: > But in the commit `feat(scyllaclient): add api for config uuid_sstable_identifiers_enabled` we can still see that `gen/agent/models/node_info` was modified while `agent.json` wasn't. After that it is indeed regenerated in a commit which modifies `agent.json`.

Thanks for pointing out, it was in different commit, mixed with other schema changes, moved it to the commit it belongs to.



> Also, looks like import:
> 
> ```go
> "github.com/scylladb/scylla-manager/swagger/gen/scylla-manager/client/operations"
> ```
> 
> is broken in `scylla_manager_clinet.go` (missing previously present v3 in the path).

That is weird I have checked it multiple times, probably broke it during rebase, file regenerated, looks good now.

File: pkg/sstable/sstable_naming.go
Comment (dkropachev) [reply pkg/sstable/sstable_naming.go]: Thanks for pointing out, updated back to what it was

File: pkg/sstable/sstable_naming.go
Comment (dkropachev) [reply pkg/sstable/sstable_naming.go]: Added

File: pkg/scyllaclient/config_client.go
Comment (Michal-Leszczynski) [reply pkg/scyllaclient/config_client.go]: I meant that checking for status 400 by using `alternatorDisabled(err)` looks strange. Now it looks good:)

File: swagger/scylla_v2.json
Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: But still we can see that commit `fix(swagger): fix files generation` modifies `gen/agent/models/node_info` without modifying `agent.json`. On the other hand, commit `feat(scyllaclient): add api for config uuid_sstable_identifiers_enabled` generates `gen/agent/models/node_info` in a wrong way.  

It also seems like generating swagger files doesn't work when checked out to `feat(scyllaclient): add api for config uuid_sstable_identifiers_enabled` generates `gen/agent/models/node_info`.  Shouldn't commit `fix(swagger): fix files generation` come first?

File: pkg/scyllaclient/config_client.go
Comment (dkropachev) [reply pkg/scyllaclient/config_client.go]: Oh, yes, I have renamed it.

File: swagger/scylla_v2.json
Comment (dkropachev) [reply swagger/scylla_v2.json:4962]: True, there was mess with commit, fixed that

Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: There are still 2 commits modifying `gen/agent/models/node_info`, and only 1 changing `agent.json`.
Let's do it in that order:
- commit fixing current problems with swagger definitions and regenerating related swagger files
- commit extending definitions with API for checking node's uuid format and regenerating related swagger files
- the rest of implementation

Let's keep changes in swagger definitions and regenerating swagger files in one commit. This approach will allow for introducing additional checks on gh actions described [here](https://github.com/scylladb/scylla-manager/pull/3461#discussion_r1272167585), but instead of doing it only on branch HEAD, we could do it for every new commit introduce in given PR.

Comment (Michal-Leszczynski) [reply swagger/scylla_v2.json:4962]: This could also help with catching things like commit `feat(scyllaclient): add api for config uuid_sstable_identifiers_enabled` does not compile.

**Review by Michal-Leszczynski — APPROVED** [review]

**Pull request merged**
